# Hypothesis Testing Exercise

#### 1. A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.
Minitab File : Cutlets.mtw

### Answer

### Null hypothesis = Mean of Unit A and Unit B are equal.
### Alternate hypothesis = Mena of Unit A and Unit B are not equal.
 

In [14]:
import numpy as np
import scipy
from scipy import stats

In [15]:
df = pd.read_csv(r"Cutlets.csv")
df.head(5)

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [16]:
stats.ttest_ind(df['Unit A'], df['Unit B'])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.47223947245995)

In [17]:
df1 = pd.read_csv(r"LabTAT.csv")
df1

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [18]:
stats.f_oneway(df1['Laboratory 1'], df1['Laboratory 2'], df1['Laboratory 3'], df1['Laboratory 4'])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

#### p-value is equal to 0.472 which is much greater than alpha value of 0.05. Hence the null hypothesis is accepted. 
 

### 2. A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
### Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
### Minitab File: LabTAT.mtw

### Answer

### Null hypothesis = Mean of Laboratory 1, Laboratory 2, Laboratory 3 and Laboratory 4 are equal.
### Alternate hypothesis = Mean of Laboratory 1, Laboratory 2, Laboratory 3 and Laboratory 4 are not equal.


In [1]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats


In [2]:
df1 = pd.read_csv(r"LabTAT.csv")
df1
stats.f_oneway(df1['Laboratory 1'], df1['Laboratory 2'], df1['Laboratory 3'], df1['Laboratory 4'])


F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

#### p-value is equal to 2.1156708949992414e-57 which is almost equal to zero and much lesser than the alpha value of 0.05. Hence the null hypothesis is rejected.  

### 3.  Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.




East
West
North
South
Males
50
142
131
70
Females
550
351
480
350




### Ho ->  All proportions are equal
### Ha ->  Not all Proportions are equal


### Check p-value
### If p-Value < alpha, we reject Null Hypothesis

### Buyer Ratio.mtw
 

### Answer

In [3]:
df = pd.read_csv(r"BuyerRatio.csv")
df.head(5)

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [4]:
#Setting Observed Values as first column
df2 = df.set_index('Observed Values')
df2

East  West  North  South
Observed Values                          
Males              50   142    131     70
Females           435  1523   1356    750

In [5]:
#Column Total
df2['Total'] = df2.sum(axis=1)
df2

East  West  North  South  Total
Observed Values                                 
Males              50   142    131     70    393
Females           435  1523   1356    750   4064

In [6]:
#Row Total
df2.loc['Total']= df2.sum()
df2

East  West  North  South  Total
Observed Values                                 
Males              50   142    131     70    393
Females           435  1523   1356    750   4064
Total             485  1665   1487    820   4457

In [7]:
#Chi-square test as sample is categorical values
stats.chi2_contingency(df2)

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.9909976552613582, dof=8, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052,
         393.        ],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948,
        4064.        ],
       [ 485.        , 1665.        , 1487.        ,  820.        ,
        4457.        ]]))

In [8]:
df3 = pd.read_csv(r"Costomer+OrderForm.csv")
df3.head(5)

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [9]:
box = df3.melt()
crosstable = pd.crosstab(box.value, box.variable, margins=True)
crosstable

variable    India  Indonesia  Malta  Phillippines   All
value                                                  
Defective      20         33     31            29   113
Error Free    280        267    269           271  1087
All           300        300    300           300  1200

In [10]:
#Chi-square test as sample is categorical values
stats.chi2_contingency(crosstable)

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.8696189580416447, dof=8, expected_freq=array([[  28.25,   28.25,   28.25,   28.25,  113.  ],
       [ 271.75,  271.75,  271.75,  271.75, 1087.  ],
       [ 300.  ,  300.  ,  300.  ,  300.  , 1200.  ]]))

### p-value is equal to 0.9909 which is much greater than alpha value of 0.05. Hence the null hypothesis is accepted and all proportions are equal. 

### 4. TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

### Minitab File: CustomerOrderForm.mtw


### Answer

### Ho ->  All centers are independent.
### Ha ->  All centers  are dependent.


In [11]:
#Import the file
df3 = pd.read_csv(r"Costomer+OrderForm.csv")
df3.head(5)


Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [12]:
#Creating a crosstable 
box = df3.melt()
crosstable = pd.crosstab(box.value, box.variable, margins=True)
crosstable


variable    India  Indonesia  Malta  Phillippines   All
value                                                  
Defective      20         33     31            29   113
Error Free    280        267    269           271  1087
All           300        300    300           300  1200

In [13]:
##Chi-square test as sample is categorical values
stats.chi2_contingency(crosstable)


Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.8696189580416447, dof=8, expected_freq=array([[  28.25,   28.25,   28.25,   28.25,  113.  ],
       [ 271.75,  271.75,  271.75,  271.75, 1087.  ],
       [ 300.  ,  300.  ,  300.  ,  300.  , 1200.  ]]))

### p-value is equal to 0.8696 which is much greater than alpha value of 0.05. Hence the null hypothesis is accepted and all centers are independent. 